### ISSUE TITLE
[clang][analyzer] unix.Malloc false positive after no-op += 0 on uintptr_t

### ISSUE THREAD

**Issue by bozicrHT** [llvm/llvm-project#220972]
**Description:**
Analyzer reports a `unix.Malloc` memory leak after an allocation pointer is converted to `uintptr_t`, subjected to the no-op operation `+= 0`, converted back to a pointer, and passed to `free()`. The value is unchanged, so `free()` receives the original allocation pointer.

**Reproducer:**
```c
#include <stdint.h>
#include <stdlib.h>

void clang_analyzer_eval(int);

void test(void)
{
    void *allocation = malloc(1);
    if (allocation == NULL)
        return;

    uintptr_t value = (uintptr_t)allocation;

    clang_analyzer_eval(value == allocation); // TRUE
    value += 0;
    clang_analyzer_eval(value == allocation); // UNKNOWN
    
    free((void *)value);
}
```
Removing line: `value += 0;` makes the warning disappear. Compiler explorer link can be found [here](https://compiler-explorer.com/z/nf4Thb99d).
